_posts/2015-06-09-Fostering-Creativity-with-Semantically-Rich-APIs.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Semantically Rich APIs
+excerpt: Fostering Creativity with Semantically Rich APIs
+layout: post
+date: 2015-06-09 00:00:00 +0200
+author: stephen
+published: true
+---
+
+In a recent post on our blog, Kyle Fuller [wrote about](http://blog.apiary.io/2015/06/04/Hyperdrive/) the benefits of decoupling semantics from implementation for building better API clients. This post aims to explore how this decoupled approach also benefits the API design process, specifically around freeing the produces and consumers to be more creative with semantically rich APIs.
+
+## The Difficulty with Common Approaches
+
+Many modern approaches in designing and building APIs couple not only clients and servers to implementation details, but we also couple thought, creativity, and shared understanding to that implementation. Coupling servers and clients is a problem that is easily seen, and in theory no one wants a coupled approach. Kyle Fuller wrote in his post:
+
+> Imagine entering an API without knowing about any implementation details, but instead just understanding the semantics of the API. We can get the implementation details of the API at run-time to allow us to change them at any point.
+
+In waiting until runtime to get these implementation details, Hyperdrive allows for building clients that are highly evolvable and don't break when things are changed on the server.
+
+Coupling hinders API evolvability—among other things—but it also hinders API design and creativity for designers and consumers. It is a subtle impedance we overlook because of the distance we've made it in HTTP APIs and because the spectrum of frameworks, libraries, and tools promote this coupled thinking.
```

Comment:
re word "the distance we've made ...". Do you mean the progress we have made in or something else?